WordPress/gutenberg
# Site Export: ensure that the export endpoint uses Gutenberg theme classes

## What?
Extending the `WP_REST_Edit_Site_Export_Controller` endpoint to use Gutenberg theme methods.

> [!NOTE]
> This PR doesn't require backporting - the code is taken from existing Core methods/functions from 6.0/5.9. The references to Core classes and functions have been changed to `_Gutenberg/gutenberg_` only.

## Why?

Like it says in the title, ensure that the export endpoint uses Gutenberg theme classes.

It makes sure that theme exports get the benefit of the latest changes to theme json and resolver.

For example, `style.background` is a new theme.json style property, but it won't be exported by the Core endpoint until version 6.6.

This endpoint hasn't had any backwards compat extensions in Gutenberg for the last few releases and probably further back so it's probably worth doing.

## How?
Copying over the public methods and swapping WP functions/methods with Gutenberg ones 😄 

The changes live in the `/lib` folder because none of the changes are backwards compat specific. Props to @andrewserong for pointing this out 😄 

Having this extension permanently in Gutenberg means that theme.json exporting will always use the latest version of the Theme JSON family of classes. 

## Testing Instructions

Add a background image to a site in the site editor. 

<img width="273" alt="Screenshot 2024-05-10 at 5 18 15 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/c90612d8-df91-4348-b7b4-173e2769d07b">


Save the site


Export your theme

<img width="272" alt="Screenshot 2024-05-10 at 5 18 23 pm" src="https://github.com/WordPress/gutenberg/assets/6458278/86d28094-8096-4760-b64b-818a4b196b89">

Check the exported theme.json - your background styles should be featured in `style.background`.

Example snippet from my exported theme.json:

```json
	"styles": {
		"background": {
			"backgroundImage": {
				"id": 26,
				"source": "file",
				"title": "pizza1",
				"url": "http://localhost:8888/wp-content/uploads/2024/05/pizza1.jpg"
			}
		},
		"color": {
			"text": "#333"
		}
	},
```
